src/Psalm/CallMap.php

```diff
@@ -4539,7 +4539,7 @@
 'kadm5_get_principals' => ['array', 'handle'=>'resource'],
 'kadm5_init_with_password' => ['resource', 'admin_server'=>'string', 'realm'=>'string', 'principal'=>'string', 'password'=>'string'],
 'kadm5_modify_principal' => ['bool', 'handle'=>'resource', 'principal'=>'string', 'options'=>'array'],
-'key' => ['mixed', 'array_arg'=>'array'],
+'key' => ['int|string|false', 'array_arg'=>'array'],
```

Comment:
This is something I'd rather keep as `mixed` (or maybe more specifically `scalar`) because otherwise it may get in the way more than anything (especially as people rarely type their array keys), so it can't be inferred much.